include/agl/detail/aglGPUMemBlockMgr.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <agl/aglGPUCommon.hpp>
+#include <container/seadPtrArray.h>
+#include <heap/seadDisposer.h>
+#include <hostio/seadHostIONode.h>
+#include <math/seadMathCalcCommon.h>
+#include <nvn/nvn_types.h>
+#include <prim/seadBitFlag.h>
+#include <prim/seadTypedBitFlag.h>
+#include <thread/seadCriticalSection.h>
+
+namespace agl::detail {
+
+typedef sead::BitFlag32 MemoryPoolDriverBitFlag;
+
+const s32 VALID_POOL_TYPE_VALUE = -1;
+const s32 cGPUAccessMask = 0xF0000000;
+const u64 cGPUPhysicalMemorySizeAlignment = 0x1000;
+
+class MemoryPoolType : MemoryPoolDriverBitFlag {
```

Comment:
Huh, N decided to inherit from a BitFlag? o.O

Reply:
Well I don't have a better way to represent that tbh...